src/amp-story-player/amp-story-player-impl.js

```diff
@@ -705,18 +727,25 @@ export class AmpStoryPlayer {
    * @param {number} storyDelta
    */
   go(storyDelta) {
+    if (storyDelta === 0) {
+      return;
+    }
     if (
-      this.currentIdx_ + storyDelta >= this.stories_.length ||
-      this.currentIdx_ + storyDelta < 0
+      !this.circularWrapping_() &&
+      this.isIndexOutofBounds_(this.currentIdx_ + storyDelta)
     ) {
       throw new Error('Out of Story range.');
     }
-    if (storyDelta === 0) {
-      return;
+    let currentStory;
+    const newIdx = this.currentIdx_ + storyDelta;
+    if (!this.isIndexOutofBounds_(newIdx)) {
+      currentStory = this.stories_[newIdx];
+    } else {
+      currentStory =
+        storyDelta > 0
+          ? this.stories_[this.currentIdx_ + storyDelta - this.stories_.length]
+          : this.stories_[this.currentIdx_ + storyDelta + this.stories_.length];
```

Comment:
This won't work for a `newIdx` that is greater than `stories_.length`. 

e.g. given values: `storyDelta = 18`, `stories_.length = 4`, `this.currentIdx_ = 0` would return `this.stories_[-14]`.

We need the algo I shared earlier:

https://codepen.io/maenrique/pen/qBbGymX?editors=0012